optuna/integration/_lightgbm_tuner/sklearn.py

```diff
@@ -1,49 +1,887 @@
+    if actual_n_features != expected_n_features:
+        raise ValueError(
+            "`n_features` must be {} but was {}.".format(expected_n_features, actual_n_features)
         )
-        super(LGBMModel, self).__init__(*args, **kwargs)
 
+    return X
 
-class LGBMClassifier(lgb.LGBMClassifier):
-    """Proxy of lightgbm.LGBMClassifier.
 
-    See: `pydoc lightgbm.LGBMClassifier`
+def check_fit_params(
+    X: TwoDimArrayLikeType,
+    y: OneDimArrayLikeType,
+    sample_weight: Optional[OneDimArrayLikeType] = None,
+    estimator: Optional[BaseEstimator] = None,
+    **kwargs: Any
+) -> Tuple[TwoDimArrayLikeType, OneDimArrayLikeType, OneDimArrayLikeType]:
+    """Check ``X``, ``y`` and ``sample_weight``.
+
+    Args:
+        X:
+            Data.
+
+        y:
+            Target.
+
+        sample_weight:
+            Weights of data.
+
+        estimator:
+            Object to use to fit the data.
+
+        **kwargs:
+            Other keywords passed to ``sklearn.utils.check_array``.
+
+    Returns:
+        X:
+            Converted and validated data.
+
+        y:
+            Converted and validated target.
+
+        sample_weight:
+            Converted and validated weights of data.
     """
+    X = check_X(X, estimator=estimator, **kwargs)
+
+    if not isinstance(y, pd.Series):
+        y = column_or_1d(y, warn=True)
+
+    _assert_all_finite(y)
+
+    if is_classifier(estimator):
+        check_classification_targets(y)
+
+    if sample_weight is None:
+        n_samples = _num_samples(X)
+        sample_weight = np.ones(n_samples)
+
+    sample_weight = np.asarray(sample_weight)
+
+    class_weight = getattr(estimator, "class_weight", None)
+
+    if class_weight is not None:
+        sample_weight *= compute_sample_weight(class_weight, y)
+
+    check_consistent_length(X, y, sample_weight)
+
+    return X, y, sample_weight
+
+
+class LGBMModel(lgb.LGBMModel):
+    """Base class for models."""
+
+    def __init__(
+        self,
+        boosting_type: str = "gbdt",
+        num_leaves: int = 31,
+        max_depth: int = -1,
+        learning_rate: float = 0.1,
+        n_estimators: int = 100,
+        subsample_for_bin: int = 200000,
+        objective: Optional[Union[Callable, str]] = None,
+        class_weight: Optional[Union[Dict[str, float], str]] = None,
+        min_split_gain: float = 0.0,
+        min_child_weight: float = 1e-03,
+        min_child_samples: int = 20,
+        subsample: float = 1.0,
+        subsample_freq: int = 0,
+        colsample_bytree: float = 1.0,
+        reg_alpha: float = 0.0,
+        reg_lambda: float = 0.0,
+        random_state: Optional[RandomStateType] = None,
+        n_jobs: int = -1,
+        silent: bool = True,
+        importance_type: str = "split",
+        study: Optional[study_module.Study] = None,
+        timeout: Optional[int] = None,
+        train_dir: Union[pathlib.Path, str] = "optgbm_info",
```

Comment:
Not a strong opinion, but how about using the keyword-only argument for these Optuna specific arguments. We would like to add some new features including the customizable step feature, and they require more arguments.

Reply:
Like `LGBMModel`, `OptunaSearchCV` has `study` and `timeout`.
Changing either class may confuse the user.

Reply:
Thank you for pointing it out. I think we can update `OptunaSearchCV` as well since it is still experimental.
But, we can work on it in a follow-up PR, so please keep the arguments as it is.